### ISSUE TITLE
Repro for Fable3

### ISSUE THREAD

**Issue by gusty** [wallymathieu/fable-plus-trait-tuple#1]
@alfonsogarciacaro this is a repro of something that works in F# but fails in Fable3

Not sure it it's a Fable issue, or a Witness Passing one.

cc: @ncave

**Comment by ncave** [comment]
@gusty Is the repro missing?

**Comment by alfonsogarciacaro** [comment]
Thanks @gusty and @wallymathieu! I will have a lool. There are indeed still some cases that make witnesses crash. In Fable we are just absorbing the error as a workaround, but @ncave has an issue in fsharp repo to track failing cases, so if any of your tests is failing because of witnesses I will add the samples to that issue.

About Async.RunSynchronously, unfortunately this is not supported by Fable as it's not possible to stop the thread in JS until something happens.

**Comment by gusty** [comment]
Cool, just to clarify, this is the repro: https://github.com/wallymathieu/fable-plus-trait-tuple/blob/main/FPlus.fs

And tests are here: https://github.com/wallymathieu/fable-plus-trait-tuple/blob/main/ExtensionsTest.fs

I tried to minimize it a bit, but then it starts working.

**Comment by wallymathieu** [comment]
Yes, I've used a polyfill for Async.RunSynchronously in Fuchu that does the job well enough (but could cause confusion). How is that usually done in Fable-applications?

https://github.com/mausch/Fuchu/blob/4cae84092d8832579ab01b68fa7781c65ee7c34f/Fuchu/Fuchu.fs#L32-L55
